### PR TITLE
Custom useatom hooks

### PR DESCRIPTION
**Pull request by LuckyPigeon**
refer to #721

## Description

Add utility functions documentation.

### CODE REVIEW

**Comment by LuckyPigeon** [comment]
@dai-shi 
This pull request contains only documentation, I'll create other PRs for each code example.

File: docs/advanced-recipes/custom-useatom-hooks.mdx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+## utils
+
+### useSelectAtom
+
+```ts
+import { useAtom } from 'jotai'
+import { selectAtom } from 'jotai/utils'
+
+export function useSelectAtom(anAtom, keyFn) {
+    return useAtom(selectAtom(anAtom, keyFn));
```
Comment (dai-shi) [inline docs/advanced-recipes/custom-useatom-hooks.mdx:14]: Please add a note that in this case `keyFn` must be stable, either define outside render or warp with `useCallback`.

File: docs/advanced-recipes/custom-useatom-hooks.mdx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+## integrations
+
+### useFocusAtom
+
+```ts
+import { useAtom } from 'jotai'
+import { focusAtom } from 'jotai/optics'
+
+export function useFocusAtom(anAtom, keyFn) {
+    return useAtom(focusAtom(anAtom, keyFn))
```
Comment (dai-shi) [inline docs/advanced-recipes/custom-useatom-hooks.mdx:49]: Please add a note that in this case `keyFn` must be stable, either define outside render or warp with `useCallback`.

**Review by ciccio423 — APPROVED** [review]

File: docs/advanced-recipes/custom-useatom-hooks.mdx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# Costuom useatom Hooks
```
Comment (dai-shi) [inline docs/advanced-recipes/custom-useatom-hooks.mdx:1]:
```suggestion
---
title: Custom useAtom hooks
nav: 5.3
---

```

**Review by dai-shi — APPROVED** [review]
Thanks for your work.

**Pull request merged**